coalib/parsing/filters/FilterBase.py

```diff
@@ -0,0 +1,20 @@
+from abc import ABC, abstractmethod
+
+
+class FilterBase(ABC):
+
+    def get_filtered_bears(self, bears, args):
+        if args:
+            bears = self.filter_bears(bears, args)
+        return bears
```

Comment:
up to you: I think it's not really necessary to have a function `get_filtered_bears` which just checks for the presence of arguments and then invokes another function in that case. I think it's simple enough to inline it into `filter_bears`. External calls look also a bit nicer with `filtered = filter_bears(...)` instead of `filtered = get_filtered_bear(...)` (which indicates a bit for me that the bears were already filtered and stored somewhere :D)